src/main/java/io/github/fablabsmc/fablabs/api/fluidvolume/v1/FluidView.java

```diff
@@ -0,0 +1,87 @@
+package io.github.fablabsmc.fablabs.api.fluidvolume.v1;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.github.fablabsmc.fablabs.api.fluidvolume.v1.containers.EntityFluidContainer;
+import io.github.fablabsmc.fablabs.api.fluidvolume.v1.containers.ItemFluidContainer;
+import io.github.fablabsmc.fablabs.api.fluidvolume.v1.containers.SidedFluidContainer;
+import io.github.fablabsmc.fablabs.api.fluidvolume.v1.volume.ImmutableFluidVolume;
+import io.github.fablabsmc.fablabs.api.fluidvolume.v1.volume.MultiFluidContainer;
+import io.github.fablabsmc.fablabs.api.fluidvolume.v1.volume.api.FluidContainer;
+import io.github.fablabsmc.fablabs.api.fluidvolume.v1.world.WorldFluidCallback;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.Entity;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Box;
+import net.minecraft.util.math.Direction;
+import net.minecraft.world.World;
+
+public final class FluidView {
+	/**
+	 * attempt to get a fluid container from the world from a specific block (or entities) for a given face.
+	 *
+	 * @param world the world
+	 * @param pos   the position
+	 * @param side  the side of the block to access
+	 * @return ImmutableFluidVolume#EMPTY if no fluid container exists for the given side
+	 */
+	public static FluidContainer getContainer(World world, BlockPos pos, Direction side) {
+		BlockState state = world.getBlockState(pos);
+		Block block = state.getBlock();
+
+		FluidContainer container = ImmutableFluidVolume.EMPTY;
+
+		if (block instanceof SidedFluidContainer) {
+			container = ((SidedFluidContainer) block).getContainer(world, state, pos, side);
+		} else {
+			SidedFluidContainer sidedContainer = WorldFluidCallback.EVENT.invoker().getContainer(world, pos);
+
+			if (sidedContainer != null) {
+				container = sidedContainer.getContainer(world, state, pos, side);
+			}
+		}
+
+		List<FluidContainer> containers = new ArrayList<>();
+
+		for (Entity entity : world.getEntities((Entity) null, new Box(pos), e -> e instanceof EntityFluidContainer)) {
+			containers.add(((EntityFluidContainer) entity).getContainer(side));
+		}
+
+		if (!containers.isEmpty()) {
+			containers.add(container);
+			return new MultiFluidContainer(containers);
+		}
+
+		return container;
+	}
+
+	/**
+	 * attempt to get a fluid container from an ItemStack.
+	 * because of the way buckets and potions are implemented, buckets and potions are ImmutableFluidVolumes, so plan accordingly!
+	 *
+	 * @param stack the stack with a count of one
+	 * @return ImmutableFluidVolume#EMPTY if no fluid container exists for the itemstack
+	 */
+	public static FluidContainer getContainer(ItemStack stack) {
+		if (stack.getCount() != 1) {
+			throw new IllegalArgumentException("ItemStack must be of size one! Hint: PlayerSlotFluidContainer");
```

Comment:
Probably not the best idea, since universal fluid cells are a popular concept.

Reply:
I could change it to remind the user that the stack is not a singular stacked item, for universal fluid cells, you'd have to ensure that the fluids are split equally, thankfully the fraction system makes this easy enough, but I'll have to make sure to include it.

Reply:
Oh wait right that could get confusing for accessing. I think it should be fine then; as long as this doesn't forbid stacking when not being accessed then it should be fine.

Reply:
done